sway/desktop/xwayland.c

```diff
@@ -284,6 +284,11 @@ static void handle_commit(struct wl_listener *listener, void *data) {
 		transaction_notify_view_ready_by_size(view,
 				surface_state->width, surface_state->height);
 	}
+
+	if (container_is_floating(view->swayc)) {
+		view_update_size(view, surface_state->width, surface_state->height);
```

Comment:
Note that this function will damage stuff and that it's not necessary because `wlr_surface` does this automatically. Maybe we should just set `view->width` and `view->height`?

Reply:
Are you sure the damage isn't necessary? What if the surface becomes smaller?

I think you'd need to set `view->swayc->current.view_{width,height}` in addition to `view->{width,height}`. The view properties are considered pending, and the current state is what's used for rendering.

However, you might not want to write them into the current state if there are transactions pending. I would try something like:

    if (view->swayc->instructions->length) {
    	transaction_notify_view_ready_by_size(view,
    			surface_state->width, surface_state->height);
    } else if (container_is_floating(view->swayc)) {
    	view_update_size(view, surface_state->width, surface_state->height);
    }

...then make `view_update_size` set both the the pending properties and current properties. Not sure about damage.

Reply:
>Are you sure the damage isn't necessary? What if the surface becomes smaller?

Nah, this is handled by `wlr_surface`. Though, it's necessary if the view has borders, so in the end I'll probablu keep it.

>I think you'd need to set view->swayc->current.view_{width,height} in addition to view->{width,height}.

The input code is using pending properties, that's why it worked. We probably need to change that.

>However, you might not want to write them into the current state if there are transactions pending.

Well I don't know what to do if transactions are in progress. Once the transaction is applied the view won't have the size we want (from `surface_state`), it'll have the size of the transaction. We probably need to queue a fake transaction instead.

Reply:
> Well I don't know what to do if transactions are in progress. Once the transaction is applied the view won't have the size we want (from surface_state), it'll have the size of the transaction.

This problem isn't specific to floating xwayland views. We ultimately rely on the client to set their size to what we configure them to. If they don't, the transaction times out and we render them incorrectly.

If we have an xwayland view in a transaction and it resizes to something other than what we tell it to, once the transaction has applied it'll resize the swayc anyway.

Reply:
> If we have an xwayland view in a transaction and it resizes to something other than what we tell it to, once the transaction has applied it'll resize the swayc anyway.

What if it's floating?

Reply:
I meant floating, and edited my comment to say so. You saw the pre-edit version.

Reply:
Ah okay, this makes more sense